web-server/src/controllers/auth/register.ts

```diff
@@ -1,60 +1,56 @@
 import { Request, Response, NextFunction } from 'express';
-
 import { AppDataSource } from 'orm/data-source';
 import { User } from 'orm/entities/users/User';
-import { CustomError } from 'utils/response/custom-error/CustomError';
-import { ErrorArray } from 'utils/response/custom-error/types';
+import { UserError } from 'utils/Errors';
 
 export const register = async (req: Request, res: Response, next: NextFunction) => {
-  const errors = new ErrorArray();
-
-  const { email, username, password } = req.body;
+  const { email, username, password, passwordConfirm } = req.body;
 
   const userRepository = AppDataSource.getRepository(User);
+  
+  const err: UserError = {};
+
   try {
+
     let user = await userRepository.findOne({ where: { email } });
 
     if (user) {
-      errors.put('email', `User with email ${email} already exists`);
+      err.email = `User with email "${email}" already exists`
     }
 
     user = await userRepository.findOne({ where: { username } });
 
     if (user) {
-      errors.put('username', `User with username ${username} already exists`);
+      err.username = `User with username "${username}" already exists`;
     }
 
+    user = new User();
+    user.email = email;
+
     try {
-      user = new User();
-      user.email = email;
-
-      try {
-        user.setUsername(username);
-      } catch (err: unknown) {
-        if (err instanceof CustomError) {
-          errors.extend(err.JSON.errors);
-        } else {
-          const customError = new CustomError(400, 'Raw', 'Error', err, errors);
-          return next(customError);
-        }
+      user.setUsername(username);
+    } catch (e) {
+      if ('username' in e) {
+        err.username = e.username;
+      } else {
+        err.raw = e;
       }
+    }
```

Comment:
I'm personally not a fan of

try {
  object.setValue(x);
} catch (e) {
  // Okay now we know x is invalid.
}

Why not:

if (validateUsername(x)) {
  user.name = x;
} else {
  // Okay username is invalid.
}

It's the same thing but it's much cleaner to look at.

Reply:
we did this po pala since the current setUsername() (in the User model) can throw its own errors/error messages, so we just did try/catch

Reply:
Yes I know. I mean take that entire function and name it "validateUsername", move it to a different file, and remove the assignment operation from it.